### PR TITLE
split lens dropdown into multiple tricks

### PR DESCRIPTION
**Pull request by Hamsda**
To give people more granular control over which parts of lens logic they know, this adds a trick per dungeon (and per MQ and vanilla variant) where lens logic is relevant. It also obviously adds a trick for lensless Wasteland.

Shadow Temple is split into before and beyond Huge Pit room, since the beginning is probably fairly easy for most people (at least non MQ), some of the platforms and targets in the later parts of the dungeon maybe not so much.

For the `Glitched World` I literally just replaced the `can_see_with_lens` occurrences in MQ dungeons with their respective tricks. Not sure if anyone is actually maintaining these 🤷 

I just added the tricks to the end of the list, no idea if that list has any order. They also have tags already for #1024 since those shouldn't cause any issues being added here.

Also updated the presets to keep the status quo (and deleted the no longer relevant MW tourney preset in the process).

For `settings_mapping.json` I am unsure how `row_span` actually works, but going from `[3,3,3]` to `[1,1,1]` produced a decent visual.

### CODE REVIEW

**Comment by r0bd0g** [comment]
I'm not sure this is a good idea just b/c of how extra complicated it makes things.

The trick for lens requirement in MQ BotW is not really necessary. The one requirement was just a small kindness for some annoying invisible enemies that I would rather see removed than to have a trick to disable.

Might as well go for the full jank and add one for Chest Minigame lol.

**Comment by AvoryGWAF** [comment]
I'd love configurable lens requirements tbh. I'm comfortable with vanilla Spirit and BotW. Vanilla shadow I have issues with the moving platform. Shadow trial is weird. I don't even know how you're supposed to do it in vanilla. I just use longshot and hovers. I'd be perfectly ok with doing chest game repeatedly. I just hate having to click required everywhere when I really only use it for MQ, that one moving platform in Shadow, Wasteland, and while I've never had to use it in shadow trial I suspect it might come up

**Comment by TreZc0** [comment]
This is something really required to close the gap between the settings that are used in current weeklys and the low tier side gigs that exist. It should have been a thing in the first place and should be added now.

As for Lens game, i don't quite see why it should ever be in logic without lens though. In any context.

**Comment by r0bd0g** [comment]
I really think the lens requirement in BotW MQ should just be removed. It never really needed to be there. It's just for some invisible enemies that can easily be targeted and only really become much of a problem on ohko, but it you're on ohko you've got bigger problems. On the other hand there are some invisible enemies guarding the basement skull that you could consider adding a lens requirement for if you really feel the current one is necessary.

I've been wondering about vanilla BotW back left bombable. Currently it doesn't ask for lens, but I've definitely seen new players struggle with that hole.

**Comment by flagrama** [comment]
[Split Lens of Truth Logic](https://trello.com/c/SvM8Nyw8/6-split-lens-of-truth-logic)

**Comment by r0bd0g** [comment]
II was thinking about this some more and I realized that the shadow split for the two lens options is actually occurring at a bit of a strange spot. I would have expected the 2nd half of shadow to start either at the moving invisible block or at the truth spinner, but it's actually somewhere in between. I think the split should start at the truth spinner. Particularly in MQ where you might be expected to progress without the Hovers, you might find yourself stuck at that spinner, falling and dying over and over. Even in vanilla you have to know to wear the Hovers to escape the pit if you guess wrong.

**Comment by Zannick** [comment]
This looks compact enough to go in now. If there's any more adjusting to be done, please follow up in another PR.

File: data/presets_default.json [hunk 1]
```diff
@@ -280,7 +210,13 @@
       "logic_rusted_switches",
       "logic_windmill_poh",
       "logic_crater_bean_poh_with_hovers",
-      "logic_forest_vines"
+      "logic_forest_vines",
```
Comment (Zannick) [inline data/presets_default.json:213]: This appears to be a change to Scrub tournament settings.

Comment (Zannick) [reply data/presets_default.json:213]: (Let me know if I should undo these changes or push as-is.)

Comment (Hamsda) [reply data/presets_default.json:213]: As discussed in Discord, these were intentional. If the preset needs to be changed in the future, it'll be in a separate PR.

**Comment by Hamsda** [comment]
Few last changes based on r0bs comments:
- I just got rid of the BotW MQ stuff
- the pit near the bombable chest in BotW probably deserves one though
- I decided to move the split for Shadow Temple tricks a bit back to the moving platform instead, since that was the point of contention from what I gathered

Can merge if everything else was alright, @Zannick

**Pull request merged**